lib/WebSocket.js

```diff
@@ -262,10 +262,10 @@ class WebSocket extends EventEmitter {
    */
   close (code, data) {
     if (this.readyState === WebSocket.CLOSED) return;
-
     if (this.readyState === WebSocket.CONNECTING) {
-      this.readyState = WebSocket.CLOSED;
-      return;
+      this._req.abort();
+      this.emit('error', new Error('closed while connecting'));
```

Comment:
Some alternatives:

- "closed before the connection is established"
- "closed before receiving a handshake response"

I have no strong opinions.